### PR TITLE
feat: Fast method for set bit calculation

### PR DESCRIPTION
**Pull request by Rishikesh-kumar-7258**
#### Description of Change
Added a fast method for set bit calculation.

Contributors guide: https://github.com/TheAlgorithms/C-Plus-Plus/CONTRIBUTING.md

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, the test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1670"></a>

### CODE REVIEW

**Comment by Rishikesh-kumar-7258** [comment]
why is this code not getting accepted?
can you tell me so I could change that...

**Comment by Rishikesh-kumar-7258** [comment]
I tried to follow the template according to that file.
since both the files address the same problem, there wasn't much to change.
I changed the algorithm and other things neccessary.

**Review by Panquesito7 — COMMENTED** [review]
> I tried to follow the template according to that file. since both the files address the same problem, there wasn't much to change. I changed the algorithm and other things neccessary.

What if we rename this file to `count_of_set_bits2.cpp`?

**Comment by Rishikesh-kumar-7258** [comment]
No problem. New name is fine.

**Comment by Panquesito7** [comment]
> No problem. New name is fine.

Feel free to change it, then. 🙂

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Can you rename the file to `count_of_set_bits_fast2.cpp`, please?

**Comment by Rishikesh-kumar-7258** [comment]
I have changed the name.  Please check it once.

File: bit_manipulation/count_of_set_bits_fast2.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * bits](https://www.geeksforgeeks.org/count-set-bits-in-an-integer/)
+ * implementation
+ */
+namespace count_of_set_bits {
+/**
+ * @brief The main function implements set bit count
+ * @param n is the number whose set bit will be counted
+ * @returns the count of the number set bit in the binary representation of `n`
+ */
+std::uint64_t countSetBits(int n) {
```
Comment (Panquesito7) [inline bit_manipulation/count_of_set_bits_fast2.cpp:41]: Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). *Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 😃

**Pull request closed without merging**